pydm/exception.py

```diff
@@ -60,25 +128,32 @@ def _add_missing_frames(tb):
     return result
 
 
-def install(use_dialog=True):
+def install(use_default_handler=True):
     """
     Install the exception hook handler.
     If use_dialog is specified, a QMessageBox will also be presented.
 
     Parameters
     ----------
-    use_dialog : bool
-        Wether or not to display a QMessageBox to the operator.
+    use_default_handler : bool
+        Whether or not to use the default handler. If not using it, users
+        must connect to the `ExceptionDispatcher.newException` signal.
     """
     global _old_excepthook
-    global _use_dialog
 
     if _old_excepthook is None:
         _old_excepthook = sys.excepthook
 
-    _use_dialog = use_dialog
+    dispatcher = ExceptionDispatcher()
+    if dispatcher.isRunning():
+        return
+    if use_default_handler:
+        handler = DefaultExceptionNotifier()
+        dispatcher.newException.connect(handler.receiveException)
```

Comment:
Seems like this should be done in the `__init__` of `DefaultExceptionNotifier` - the notifier should be in control itself so external code never accidentally connects `receiveException` more than once.